### PR TITLE
drop ycp debugger

### PR DESCRIPTION
**Pull request by jreidinger**
I check usage in other yast bindings and no reference

### CODE REVIEW

**Comment by mvidner** [comment]
Travis is failing because it cannot link y2base. How did you test that you did not break things?

**Comment by jreidinger** [comment]
@mvidner manual rake osc:build works, for travis I am not sure about root of issue

**Comment by mvidner** [comment]
Hmm, I don't know either.
Anyway, what is the reason for this PR? I can think of some but let's hear yours :)

**Comment by jreidinger** [comment]
@mvidner well, two main reasons

1. yast2-core-debugger is confusing package as it won't work not trigger byebug we use
2. debugger adds unnecessary complexity to code, I notice it when doing ErrorNamespace where is some enabling of tracking for debugger and for it some special conditions needed

**Comment by mvidner** [comment]
OK, so let's use this .changes message:
- Dropped the YCP debugger in yast2-core-debugger not to confuse people who want to use the Byebug based Ruby debugger

Please mention https://github.com/yast/yast.github.io/blob/master/doc/debugging.md and its rendering http://yastgithubio.readthedocs.io/en/latest/debugging/ in the commit message

File: liby2/src/genericfrontend.cc [hunk 1]
```diff
@@ -657,42 +651,19 @@ main (int argc, char **argv)
 
 // FIXME the whole option parsing sucks **** !
 
-    // Check, if debugger should be enabled
+    // old backward compatibility stuff
```
Comment (mvidner) [inline liby2/src/genericfrontend.cc]: I think we should simply drop this. If anyone were still using `--debugger` they would expect the YCP thing. We are removing it so let's remove this too.

Comment (mvidner) [inline liby2/src/genericfrontend.cc]: It is also mentioned in https://github.com/yast/yast.github.io/blob/master/doc/environment-variables.md

Comment (jreidinger) [inline liby2/src/genericfrontend.cc]: ok, fine for me. I just keep it to be compatible, but if it is not needed, drop is easy.

**Comment by lslezak** [comment]
For the Travis issue see https://github.com/yast/yast-core/pull/117, switching to Docker at Travis should help here.

**Comment by lslezak** [comment]
I have rebased the PR on the latest `master` to use Docker at Travis. Now it is green :white_check_mark:

**Pull request merged**